### PR TITLE
Return the success value of page.waitFor

### PR DESCRIPTION
**Pull request by JoelEinbinder**
I will update api.md in this PR after #318 lands.

### CODE REVIEW

**Comment by aslushnikov** [comment]
I'm not sure I follow. Why is this needed?

**Comment by JoelEinbinder** [comment]
`waitForSelector` should return an ElementHandle. `waitForFunction` should return an ElementHandle if the function resolves to an Element. For simplicity and consistency, `waitForFunction` should just return its success value.

**Comment by aslushnikov** [comment]
But in this case, the success value will always be `true`?

**Comment by JoelEinbinder** [comment]
It is always truthy, but it isn't necessarily `true`. Look at the test.

File: test/test.js [hunk 1]
```diff
@@ -332,6 +332,12 @@ describe('Page', function() {
       expect(error).toBeTruthy();
       expect(error.message).toContain('Cannot poll with non-positive interval');
     }));
+    it('should return the success value', SX(async function() {
+      const watchdog = page.waitForFunction(() => window.__FOO);
+      let success = await page.evaluate(() => window.__FOO = 'hit');
+      await watchdog;
+      expect(success).toBe('hit');
```
Comment (aslushnikov) [inline test/test.js:339]: This probably should be
```js
expect(await watchdog).toBe('hit')
```

Comment (JoelEinbinder) [reply test/test.js:339]: Yep. Good catch.

**Comment by aslushnikov** [comment]
I see now. Prior to landing this, let's discuss our element handle plan with @pavelfeldman / @dgozman so that we're all on the same page.

**Comment by aslushnikov** [comment]
Do you want this in now or you'll wait for #382 to land first?

**Comment by aslushnikov** [comment]
Closing this in favor of object handles.

**Pull request closed without merging**